code/surface_area.py

```diff
@@ -0,0 +1,421 @@
+                print(f"✓ Number of Vertices: {num_vertices}")
+
+                # Statistical measures
+                data_min = np.min(data)
+                data_max = np.max(data)
+                data_mean = np.mean(data)
+
+                # Validate data ranges (surface areas should be positive)
+                if data_min >= 0:
+                    validation_results["Minimum"] = True
+                    print(f"✓ Minimum: {data_min:.6f} (valid non-negative value)")
+                else:
+                    validation_results["Minimum"] = False
+                    print(f"✗ Minimum: {data_min:.6f} (should be non-negative)")
+
+                if data_max > 0:
+                    validation_results["Maximum"] = True
+                    print(f"✓ Maximum: {data_max:.6f} (valid positive value)")
+                else:
+                    validation_results["Maximum"] = False
+                    print(f"✗ Maximum: {data_max:.6f} (should be positive)")
+
+                if data_mean > 0:
+                    validation_results["Mean"] = True
+                    print(f"✓ Mean: {data_mean:.6f} (valid positive value)")
+                else:
+                    validation_results["Mean"] = False
+                    print(f"✗ Mean: {data_mean:.6f} (should be positive)")
+
+            else:
+                validation_results["Number of Vertices"] = False
+                validation_results["Minimum"] = False
+                validation_results["Maximum"] = False
+                validation_results["Mean"] = False
+                print("✗ No data found in the metric file")
+
+        else:
+            validation_results["Maps to Surface"] = False
+            validation_results["Map Name"] = False
+            validation_results["Number of Vertices"] = False
+            validation_results["Minimum"] = False
+            validation_results["Maximum"] = False
+            validation_results["Mean"] = False
+            print("✗ Maps to Surface: false (no data arrays found)")
+
+        # Overall validation result
+        all_passed = all(validation_results.values())
+
+        if all_passed:
+            print("✓ All field validations passed")
+        else:
+            failed_checks = [k for k, v in validation_results.items() if not v]
+            print(f"✗ Failed validations: {', '.join(failed_checks)}")
+
+        return all_passed
+
+    except Exception as e:
+        print(f"✗ Validation failed: {e}")
+        return False
+
+
+def test_output_file(output_metric: Path, validate: bool = True):
+    """
+    Test if the output metric file exists and has valid data fields.
+    """
+    # First run the niwrap file_information to ensure it works
+    try:
+        wb.file_information(str(output_metric))
+        print("✓ niwrap file information executed successfully")
+    except Exception as e:
+        print(f"✗ niwrap file information failed: {e}")
+        return False
```

Comment:
I'm not sure we still need this if we are attempting to validate the same thing now with `nibabel`. Maybe you can walk me through your thoughts here before we decide to keep / remove this?